### ISSUE TITLE
[css-forms-1] Names of <input type=range> / <input type=checkbox switch> pseudo-elements

### ISSUE THREAD

**Issue by annevk** [w3c/csswg-drafts#9830]
I decided to split this from #4410 as that discussion is already quite long.

In WebKit we accidentally implemented the new pseudo-elements as `::thumb` and `::track`. They are not shipped and can be renamed, but when discussing it internally none of us really liked the longer prefixed names. Also, when it was revealed we implemented these pseudo-elements, nobody noticed they had the wrong names. @nt1m discovered it while auditing pseudo-elements in general.

Selectors are typically short words, without thematic grouping. E.g., it's :playing, not :media-playing. They get context from usage. E.g., input::thumb.

And for a switch it's also not at all a "slider" thumb. It's just the thumb. It slides while animating, but it's not like you can hold it in any position that's not on or off.

If there's ever a need for another thumb/track-like pseudo-element that can have a longer name.

**Comment by yisibl** [comment]
> RESOLVED: Add ::slider-thumb, ::slider-track
RESOLVED: Define these two pseudos as siblings of each other
RESOLVED: Apply them to <input type=range>, switch, <progress>, and <meter>
RESOLVED: Add ::slider-fill when relevant
RESOLVED: track, fill, thumb are in the same tree order as their usual painting order

https://github.com/w3c/csswg-drafts/issues/4410#issuecomment-1720875895

I found a problem with the name of the WebKit implementation, but I thought the CSSWG had discussed it elsewhere and removed the `slider-` prefix from the name.😂

Overall, I like the shorter names.

**Comment by yisibl** [comment]
In addition, we need to clarify whether these pseudo-elements should support the content property. See https://github.com/openui/open-ui/issues/959#issuecomment-1866027222

**Comment by lukewarlow** [comment]
I was under the impression these were the resolved names at least for input type range?

**Comment by fantasai** [comment]
My concern with the shorter names is that some of them are very generic, e.g. `::track` especially and even `::thumb`. Imho grouping them together makes it clearer what they're referring to.

**Comment by lukewarlow** [comment]
I agree that `::thumb` might be a bit generic especially when we already have the (admittedly non-standard) `::-webkit-scrollbar-thumb`. But `slider` also doesn't seem like the right prefix if we want these to apply to `<meter>` and `<progress>`, and is a bit odd even for `<input type=checkbox switch>`.
